### DISCUSSION TITLE
What is NN_theta(X)?

### DISCUSSION THREAD

**Discussion by benpymer** [Part-II-Computational-Physics/.github#80] [General]
I don't know if I'm being dumb, but I feel like I would be able to do most of this exercise, except for the fact that I've no idea how to construct my NN_theta(X) function in task 1. I went to the lecture notes and saw that you could create a layer which multiplied your input by a weight matrix and then added a bias and then combine those functions, but I've no idea how to combine that with the neural net that's been initialised from micrograd, and I'm kind of struggling to understand exactly what micrograd does and how the micrograd library works apart from the fact that it has a neat gradient finding function and can do .relu to stuff which is handy for AI. Does anyone have advice on constructing NN_theta?

**Comment by ceciliegl** [comment]
Hi @benpymer !

The code
model = MLP(2, [n_neurons, n_neurons, 1])
in the function initialise(n_neurons) constructs the neural network NN_theta with parameters theta (weights and biases). The scores = NN_theta(Xdat) are found through
inputs = [list(map(Value, xrow)) for xrow in Xdat]
scores = list(map(model, inputs))
which is already implemented in loss(Xdat,ydat,model).

The parameters theta of the neural network can be reached through model.parameters().

Hope this helps. Let me know if you have any more questions!

**Reply by benpymer** [reply]
Thanks so much! That makes a lot more sense! I was struggling to understand what all the code did, but I think I've now got it working!
